qa/zcash/smoke_tests.py

```diff
@@ -0,0 +1,808 @@
+    ('4cc', True,  True ), # Send funds from t-addr to Sapling z-addr using z_mergetoaddress
+    ('4dd', True,  True ), # Send funds from t-addr and Sprout z-addr to Sprout z-addr using z_mergetoaddress
+    ('4ee', True,  True ), # Send funds from t-addr and Sapling z-addr to Sapling z-addr using z_mergetoaddress
+    ('4ff', True,  True ), # Send funds from Sprout z-addr and Sprout z-addr to Sprout z-addr using z_mergetoaddress
+    ('4gg', True,  True ), # Send funds from  Sapling z-addr and Sapling z-addr to Sapling z-addr using z_mergetoaddress
+    # Wallet
+    ('5a', True, True), # After generating multiple z-addrs, run z_listaddresses
+    ('5b', True, True), # Run z_validateaddress with a Sprout z-addr
+    ('5c', True, True), # Run z_validateaddress with a Sapling z-addr
+    ('5d', True, True), # After a transaction, run z_listunspent
+    ('5e', True, True), # After a transaction, run z_listreceivedbyaddress
+    ('5f', True, True), # After a transaction, run z_getbalance
+    ('5g', True, True), # After a transaction, run z_gettotalbalance
+    ('5h', True, True), # Run z_exportkey using a Sprout z-addr
+    ('5i', True, True), # Run z_importkey using the zkey from a Sprout z-addr
+    ('5j', True, True), # Run z_exportkey using a Sapling z-addr
+    ('5k', True, True), # Run z_importkey using the zkey from a Sapling z-addr
+    ('5l', True, True), # Run z_exportwallet
+    ('5m', True, True), # Run z_importwallet
+    ('5n', True, True), # Run z_shieldcoinbase
+    ('5o', True, True), # Run getwalletinfo
+    # Network
+    ('6a', True, True), # Run getpeerinfo
+    ('6b', True, True), # Run getnetworkinfo
+    ('6c', True, False), # Run getdeprecationinfo
+    ('6d', True, True), # Run getconnectioncount
+    ('6e', True, True), # Run getaddednodeinfo
+    # Mining
+    ('7a', True, True), # Run getblocksubsidy
+    ('7b', True, True), # Run getblocktemplate
+    ('7c', True, True), # Run getmininginfo
+    ('7d', True, True), # Run getnetworkhashps
+    ('7e', True, True), # Run getnetworksolps
+]
+
+TIME_STARTED = datetime.datetime.now()
+
+#
+# Test helpers
+#
+
+def run_cmd(results, case, zcash, name, args=[]):
+    print('----- %s -----' % (datetime.datetime.now() - TIME_STARTED))
+    print('%s $ zcash-cli %s %s' % (
+        case.ljust(3),
+        name,
+        ' '.join([str(arg) for arg in args],
+    )))
+    try:
+        res = zcash.__getattr__(name)(*args)
+        print(res)
+        print()
+        if results is not None and len(case) > 0:
+            results[case] = True
+        return res
+    except RpcException as e:
+        print('ERROR: %s' % str(e))
+        if results is not None and len(case) > 0:
+            results[case] = False
+        return None
+
+def wait_for_balance(zcash, zaddr, expected=None, timeout=900):
+    print('Waiting for funds to %s...' % zaddr)
+    unconfirmed_balance = Decimal(zcash.z_getbalance(zaddr, 0)).quantize(Decimal('1.00000000'))
```

Comment:
This needs to be updated to account for the change to add transactions to the mempool asynchronously. `time.sleep(5)` before making this call to be on the safe side?